apps/06.1Checkers.js

```diff
@@ -3,75 +3,190 @@
+    //create the array thats going to hold the coordinates of checkers
+    this.checkers = [];
+    this.createCheckers = function () {
+            //create two arrays for initial positions of black and white checkers
+    var whitePositions = [];
+    var blackPositions = [];
+            //fill the two initial positions arrays with the coordinates of the checkers 
+    var j = 0;
+    while (j < 3) {
+    for (var i = 1; i <= 7; i = i + 2) {
+    whitePositions[j] = [0, i];
+    whitePositions[j + 8] = [2, i];
+    blackPositions[j + 4] = [6, i];
+    j++;
+}
+    j = 0;
+    for (var i = 0; i < 7; i = i + 2) {
+    whitePositions[j + 4] = [1, i];
+    blackPositions[j] = [5, i];
+    blackPositions[j + 8] = [7, i];
+    j++;
+}
+}
+            //create checkers and put the instances into the grid and into an array checkers
+    for (i = 0; i < 12; i++) {
+    this.checker = new Checker("white");
+    var x = whitePositions[i][0];
+    var y = whitePositions[i][1];
+    this.grid[x][y] = new Checker("white");
+    this.checkers.push(this.grid[x][y]);
+            }
+    for (i = 0; i < 12; i++) {
+    this.checker = new Checker("black");
+    var x = blackPositions[i][0];
+    var y = blackPositions[i][1];
+    this.grid[x][y] = new Checker("black");
+    this.checkers.push(this.grid[x][y]);
+}
 }
+        //helper function to return the instance in the grid at the correct row, column
+    this.selectChecker = function (row, column) {
+    return this.grid[row][column];
+        }
+        //this checks for a winner
+    this.checkForWinner = function () {
+        //using some, see if a value exists in the array
+    if (!this.checkers.some(function (val) {
+    return val.symbol === String.fromCharCode(0x125CB);
+})) {
+            //if it does show the winner
+    console.log("black wins!");
+}
+        //do the same as above, but for white
+    return val.symbol === String.fromCharCode(0x125CF);
+})) {
+    console.log("white wins!");
+}
+        //quit the game when there is a winner
+    process.exit();
+}
+
+    //kill a checker method
+    this.killChecker = function (position) {
+        //select the instance you want to kill
+    var checkerToKill = this.selectChecker(position[0], position[1]);
+        //find where this instance is in the array of checkers and splice it from the array
+    var indexToKill = this.checkers.indexOf(checkerToKill);
+    if (indexToKill > -1) {
+    this.checkers.splice(indexToKill, 1);
+}
+        //set the grid position to null
+    this.grid[position[0]][position[1]] = null;
+}
+
+}
+
+
 function Game() {
+    //create a new board instance
+    this.board = new Board();
+    //create the grid and the initial set of checkers
+    this.start = function () {
+        this.board.createGrid();
+        this.board.createCheckers();
+    };
+    //move a checker
+    this.moveChecker = function (start, end) {
+        //cheat code included "white wins" or "black wins"
+        if (end === "wins") {
```

Comment:
What game scenerio will the `end` argument === "wins"?


`checkForWinner();` should only run after the `this.killChecker()` method is allowed to run it's logic.